src/main/java/com/employeesService/EmployeesService/service/EmployeeServiceImpl.java

```diff
@@ -1,28 +1,59 @@
 package com.employeesService.EmployeesService.service;
 
+import com.employeesService.EmployeesService.exception.DepartmentNotFoundException;
 import com.employeesService.EmployeesService.exception.EmployeeNotFoundException;
+import com.employeesService.EmployeesService.model.Department;
+import com.employeesService.EmployeesService.model.DepartmentEmployee;
+import com.employeesService.EmployeesService.model.DepartmentEmployeeDTO;
 import com.employeesService.EmployeesService.model.Employee;
+import com.employeesService.EmployeesService.repository.DepartmentRepository;
 import com.employeesService.EmployeesService.repository.EmployeeRepository;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
 
+import java.time.LocalDate;
 import java.util.List;
+import java.util.stream.Collectors;
 
 @Service
 public class EmployeeServiceImpl implements EmployeeService {
 
     private final EmployeeRepository employeeRepository;
+    private final DepartmentRepository departmentRepository;
 
     @Autowired
-    public EmployeeServiceImpl(EmployeeRepository employeeRepository) {
+    public EmployeeServiceImpl(EmployeeRepository employeeRepository, DepartmentRepository departmentRepository) {
         this.employeeRepository = employeeRepository;
+        this.departmentRepository = departmentRepository;
     }
 
     @Override
     public Employee save(Employee employee) {
         return employeeRepository.save(employee);
     }
 
+    @Override
+    @Transactional
+    public Employee addEmployeeToDepartment(String deptNo, Long empNo, LocalDate fromDate) {
+        Department department = this.departmentRepository.findById(deptNo).orElseThrow(() -> new DepartmentNotFoundException(deptNo));
+        Employee employee = this.employeeRepository.findById(empNo).orElseThrow(() -> new EmployeeNotFoundException(empNo));
+        department.addEmployee(employee, fromDate);
+        this.departmentRepository.save(department);
+        return employee;
+    }
+
+    @Override
+    public List<DepartmentEmployeeDTO> listAllEmployeesPerDepartment(String deptNo) {
+        Department department = this.departmentRepository.findByIdWithEmployees(deptNo).orElseThrow(() -> new DepartmentNotFoundException(deptNo));
+        return department.getEmployees().stream().map(this::convertToDTO).collect(Collectors.toList());
+    }
+
+    private DepartmentEmployeeDTO convertToDTO(DepartmentEmployee employee) {
+        DepartmentEmployeeDTO departmentEmployeeDTO = new DepartmentEmployeeDTO(employee.getFromDate(), employee.getToDate(), employee.getEmployee());
+        return departmentEmployeeDTO;
+    }
```

Comment:
I prefer to have this conversion method into Controller, the service should return the entity classes usually